controller.go

```diff
@@ -337,6 +341,14 @@ func (c *controller) clusterAgentInit() {
 			// service bindings
 			c.agentClose()
 			c.cleanupServiceBindings("")
+
+			c.Lock()
+			if c.agentStopDone != nil {
+				close(c.agentStopDone)
+				c.agentStopDone = nil
```

Comment:
The shutdown of the agent is triggered by `SetClusterProvider(nil)`.
If the caller needs to do extra networking programming which requires the agent to be down, then it has to invoke the agentStopWait. That is really only for the use case of the ingress cleanup.